src/mobile-token/use-token-toggle-details.ts

```diff
@@ -1,36 +1,43 @@
-import {useEffect, useState} from 'react';
 import {useMobileTokenContextState} from '@atb/mobile-token';
+import { useIsFocused } from '@react-navigation/native';
+import {useQuery} from '@tanstack/react-query';
 
-const useTokenToggleDetails = (shouldFetchTokenDetails: boolean) => {
-  const [shouldShowLoader, setShouldShowLoader] = useState<boolean>(false);
-  const [toggleLimit, setToggleLimit] = useState<number | undefined>();
-  const [maxToggleLimit, setMaxToggleLimit] = useState<number | undefined>();
+/**
+ * 10 seconds stale/cache time to allow user get latest result in case they switch their phone/t:card
+ */
+const ONE_MINUTE = 1000 * 60;
+export const GET_TOKEN_TOGGLE_DETAILS_QUERY_KEY = 'getTokenToggleDetails';
 
-  const {getTokenToggleDetails} = useMobileTokenContextState();
+const useTokenToggleDetails = () => {
+  const {getTokenToggleDetails, deviceInspectionStatus} = useMobileTokenContextState();
+  const screenHasFocus = useIsFocused();
+  const shouldFetchTokenDetails =
+    screenHasFocus && deviceInspectionStatus !== 'loading';
+  return useQuery({
+    enabled: shouldFetchTokenDetails,
+    queryKey: [GET_TOKEN_TOGGLE_DETAILS_QUERY_KEY],
+    queryFn: async () => {
+      let toggleLimit: number | undefined;
+      let maxToggleLimit: number | undefined;
 
-  useEffect(() => {
-    (async function () {
-      if (shouldFetchTokenDetails) {
-        setShouldShowLoader(true);
-        const toggleToggleDetails = await getTokenToggleDetails();
-        if (toggleToggleDetails) {
-          const {toggleMaxLimit, toggledCount} = toggleToggleDetails;
-          if (toggleMaxLimit) {
-            if (toggleMaxLimit >= toggledCount) {
-              setToggleLimit(toggleMaxLimit - toggledCount);
-            } else {
-              //We can end up here if we decide to reduce max limit value in firestore
-              setToggleLimit(0);
-            }
+      const toggleToggleDetails = await getTokenToggleDetails();
```

Comment:
I guess you meant to call this `tokenToggleDetails`?

Reply:
it is the name of the variable before refactor, but maybe it should be `tokenToggleDetails` as you noticed.